ui/src/app/shared/api/schedule.service.ts

```diff
@@ -37,6 +37,12 @@ export class ScheduleService {
       .get<any>(`${url}?platform=${platform}`)
       .pipe(
         map(SchedulePage.parse),
+        map((page: SchedulePage) => {
```

Comment:
That parse method already tries to use platform so it might work better like this:
```
      .pipe(
        map((input) => {
          input.platform = platform;
          return input;
        }),
        map(SchedulePage.parse),
        catchError(ErrorUtils.catchError)
      );

```

Reply:
Though I kinda hate to modify input like this but we always get to these issues when using `any` type which makes contract between server response vs. data in a page a bit obscure.